pkg/trace/writer/stats.go

```diff
@@ -136,6 +172,13 @@ func (w *StatsWriter) SendPayload(p *stats.Payload) {
 	sendPayloads(w.senders, req)
 }
 
+func (w *StatsWriter) sendPayloads() {
+	for _, p := range w.payloads {
+		w.SendPayload(p)
+	}
+	w.payloads = []*stats.Payload{}
```

Comment:
Maybe it's better to do this instead of re-allocating a new slice
```suggestion
	w.payloads = w.payloads[:0]
```